### DISCUSSION TITLE
How to deal with bars without bid and ask?

### DISCUSSION THREAD

**Discussion by shiniantianlang** [nautechsystems/nautilus_trader#695] [Q&A] [answered]
Thanks so much for your work for providing a fantastic cython based trading system!
I have noticed that the K-line data used in your example is including bid and ask, and it is also specified in the `process_bar_data` function. However, my current bars only include open, high, low, close, and no bid and ask. So, how can the data be applied to this system?

**Comment by cjdsellers** [comment, marked as answer]
Hi @shiniantianlang, thanks so much for the kind feedback!

Basically it would help to know what the price type of the data is, we may be able to infer this from the source (e,g. most Crypto bar data will be based on `LAST` price or trades). So in this case I would use a `LAST` price type, as you don't know if its the bid or ask side.

**Reply by shiniantianlang** [reply]
Thanks for your quick response which gives me the confidence to continue to study the system.
